python/src/spark_rapids_ml/tuning.py

```diff
@@ -134,3 +146,32 @@ def singePassTask(
             )
 
         return self._copyValues(model)
+
+    @staticmethod
+    def _is_python_params_instance(metadata: Dict[str, Any]) -> bool:
+        # If it's not python module, pyspark will load spark_rapids_ml.tuning.CrossValidator
+        # from JVM package. So we need to hack here
+        return metadata["class"].startswith(("pyspark.ml.", "spark_rapids_ml."))
+
+    @classmethod
+    def load(cls, path: str) -> "CrossValidator":
+        orig_is_python_params_instance = DefaultParamsReader.isPythonParamsInstance
+        try:
+            # Replace isPythonParamsInstance
+            setattr(
+                DefaultParamsReader,
+                "isPythonParamsInstance",
+                CrossValidator._is_python_params_instance,
+            )
+            cv_pyspark = super().load(path)
+            cv = cls()
+            cv_pyspark._copyValues(cv)
+        finally:
+            # Must restore to the original isPythonParamsInstance
```

Comment:
Why is this necessary?

Reply:
Yeah, From the function we customized, we don't need to restore to the original function. But again, I really don't want to hack pyspark internal things. To be honest, I don't know if this hacked function will have other impacts on the pyspark.